### PR TITLE
add algo descartes(01AI)

### PR DESCRIPTION
**Pull request by xiaoming-01ai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ Evaluated
 * [pgvector](https://github.com/pgvector/pgvector) ![https://img.shields.io/github/stars/pgvector/pgvector?style=social](https://img.shields.io/github/stars/pgvector/pgvector?style=social)
 * [RediSearch](https://github.com/redisearch/redisearch) ![https://img.shields.io/github/stars/redisearch/redisearch?style=social](https://img.shields.io/github/stars/redisearch/redisearch?style=social)
   * [pg_embedding](https://github.com/neondatabase/pg_embedding) ![https://img.shields.io/github/stars/pg_embedding/pg_embedding?style=social](https://img.shields.io/github/stars/neondatabase/pg_embedding?style=social)
+* [Descartes(01AI)](https://github.com/01-ai/Descartes.git)
```
Comment (maumueller) [inline README.md:49]: This link doesn't work.

File: ann_benchmarks/algorithms/descartes/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM ann-benchmarks
+
+RUN apt update
+RUN apt install -y git cmake g++ python3 python3-setuptools python3-pip libblas-dev liblapack-dev
+RUN pip3 install wheel pybind11==2.5.0
+
+WORKDIR /home/app
+RUN git clone https://github.com/xiaoming-01ai/pydescartes.git
+RUN cp pydescartes/py01ai.cpython-310-x86_64-linux-gnu.so  ./py01ai.cpython-310-x86_64-linux-gnu.so
```
[the remark this answers is not in the document]
Comment (richardllin) [reply ann_benchmarks/algorithms/descartes/Dockerfile:9]: @maumueller Your understanding is correct.

[the remark this answers is not in the document]
Comment (xiaoming-01ai) [reply ann_benchmarks/algorithms/descartes/Dockerfile:9]: @maumueller  Yes, we are not open source yet.

File: README.md (hunk 1 above)
Comment (xiaoming-01ai) [reply README.md]: @maumueller  OK, We remove the link  and add it back after the product is released in the future.

Comment (maumueller) [reply README.md]: It wasn't my intention to remove it. I think you are pointing to the wrong github repo in the link, so please just edit.

File: ann_benchmarks/algorithms/descartes/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (maumueller) [reply ann_benchmarks/algorithms/descartes/Dockerfile:9]: @erikbern What is your opinion on including non-open-source libraries?

[the remark this answers is not in the document]
Comment (richardllin) [reply ann_benchmarks/algorithms/descartes/Dockerfile:9]: @maumueller @erikbern We've observed that `qstngt` (not open source) is also included in the list, so we believe that `descartes` should also be considered :)

**Comment by maumueller** [comment]
Thanks for the contribution!

**Pull request merged**